### PR TITLE
&fkiraly [ENH] MSRE% forecasting metric

### PR DESCRIPTION
**Pull request by jusssch**
In this PR i added the Mean Squared Error Percentage (MSRE%) forecasting metric as described in #4751.

### CODE REVIEW

**Review by yarnabrina — CHANGES_REQUESTED** [review]
Thanks for you contribution, but few changes look odd. Can you kindly take a look at the comments below?

**Review by fkiraly — CHANGES_REQUESTED** [review]
Very nice!

* unrelated, but blocking - it seems you accidentally mixed changes from your Arsenal related branch into this. Kindly ensure these are not on this branch. A "quick and hacky" fix would be to copy over the file from `main`. As we squash pull requests, that will not cause a revert.
* could you kindly add docstrings?
* `MeanAbsoluteErrorPercentage` seems incorrect. Did you mean to put the code under `evaluate`?
* `MeanSquaredErrorPercentage` looks mostly correct! Thouch, could you kindly remove the "time 100" to have unitless output, consistent with other metrics?
* the jackknife formula for `_evaluate_by_index` in the `square_root` case does not seem correct. If you are not certain, simply call `super` in the `self.square_root` case. It might also not be correct in the non-squared case, since the denominator is not constant but depends on the data too.
    * if you are not sure about this, you can remove `_evaluate_by_index` so we use the correct (but inefficient) default

File: sktime/classification/kernel_based/_arsenal.py
[the remark this answers is not in the document]
Comment (jusssch) [reply sktime/classification/kernel_based/_arsenal.py:1]: Yes, this was by accident! I removed them.

**Comment by fkiraly** [comment]
FYI, this PR seems to fail formatting conventions, see here how to set up automatic code formatting:
https://www.sktime.net/en/latest/developer_guide/coding_standards.html

File: sktime/performance_metrics/forecasting/_classes.py [hunk 1]
```diff
@@ -1888,6 +1888,178 @@ def get_test_params(cls, parameter_set="default"):
...
+
+        Returns
+        -------
+        loss : float or pd.Series
+            The calculated Mean Squared Error Percentage.
+            - If `multioutput='raw_values'`, returns a Series with the MSPE for
+            each output.
+            - Otherwise, returns a scalar value representing the aggregated MSPE.
+
+        Notes
```
Comment (fkiraly) [inline sktime/performance_metrics/forecasting/_classes.py:1949]: the notes from the private docstrings will not be displayed to the user or the online reference. Explanation should be in the class docstring, this ends up in the API reference (if you add the estimator to `api_reference/performance_metrics.rst`).

Can you kindly move the explanation up there?

File: sktime/performance_metrics/forecasting/_classes.py [hunk 2]
```diff
@@ -1888,6 +1888,178 @@ def get_test_params(cls, parameter_set="default"):
...
+        -----
+        The Mean Squared Error Percentage (MSE%) is calculated as:
+
+        .. math::
+            \\text{MSE%} = \\frac{ \\frac{1}{n} \\sum_{i=1}^{n} (y_i - \\hat{y}_i)^2 }
```
Comment (fkiraly) [inline sktime/performance_metrics/forecasting/_classes.py:1954]: should the left hand side depend on y and y-hat?

File: sktime/performance_metrics/forecasting/_classes.py [hunk 3]
```diff
@@ -1888,6 +1888,178 @@ def get_test_params(cls, parameter_set="default"):
...
+        where:
+        - \\( y_i \\) are the true values,
+        - \\( \\hat{y}_i \\) are the predicted values,
+        - \\( n \\) is the number of observations.
+
+        If `square_root` is set to True, the Root Mean Squared Error Percentage (RMSPE)
+        is computed:
+
+        .. math::
+            \\text{RMSPE} = \\sqrt{ \\text{MSPE} }
```
Comment (fkiraly) [inline sktime/performance_metrics/forecasting/_classes.py:1966]: do you want to use consistently MSPE and RMSPE, or the "percent" sign, both times?

File: sktime/performance_metrics/forecasting/_classes.py [hunk 4]
```diff
@@ -1888,6 +1888,178 @@ def get_test_params(cls, parameter_set="default"):
...
+            rmsep_jackknife = (sum - rmsep) / (n - 1)
+        else:
+            rmsep = raw_values_mse / raw_values_p
+            sum = rmsep.sum(axis=0)
+            rmsep_jackknife = (sum - rmsep) / (n - 1)
+
+        pseudo_values = n * rmsep - (n - 1) * rmsep_jackknife
+        pseudo_values = self._get_weighted_df(pseudo_values, **kwargs)
+
+        if isinstance(multioutput, str):
```
Comment (VascoSch92) [inline sktime/performance_metrics/forecasting/_classes.py:2030]: Do you need this check? `multiouput` is always a string right?

File: sktime/performance_metrics/forecasting/_classes.py (hunk 3 above)
Comment (jusssch) [reply sktime/performance_metrics/forecasting/_classes.py:1966]: I'm changing it to always be ``MSE%`` (Mean Squared Error Percentage) and ``RMSE%`` (Root Mean Squared Error Percentage)

File: sktime/performance_metrics/forecasting/_classes.py (hunk 2 above)
Comment (jusssch) [reply sktime/performance_metrics/forecasting/_classes.py:1954]: I don't really understand the question. Isn't this how MSE% / RMSE% is defined in #4751? See the following:

$\text{MSE\\%} = \frac{\sqrt{\frac{1}{n} \sum_{i=1}^{n} (y_i - \hat{y_i})^2}}{\sum_{i=1}^{n} \frac{y_i}{n}}$

Where the root can be added / removed by setting ``square_root``.

File: sktime/performance_metrics/forecasting/_classes.py (hunk 4 above)
Comment (jusssch) [reply sktime/performance_metrics/forecasting/_classes.py:2030]: Yes, you are right! Removing it...

Comment (fkiraly) [reply sktime/performance_metrics/forecasting/_classes.py]: hi @VascoSch92, how have you been doing? Playing with the symmetric group, I see?

Comment (VascoSch92) [reply sktime/performance_metrics/forecasting/_classes.py]: Hey, 

the last 6 months were pretty busy :-( 

I should join the discord channel of sktime...

Comment (fkiraly) [reply sktime/performance_metrics/forecasting/_classes.py]: https://discord.com/invite/54ACzaFsn7 😃

File: sktime/performance_metrics/forecasting/_classes.py (hunk 2 above)
Comment (fkiraly) [reply sktime/performance_metrics/forecasting/_classes.py:1954]: * this is RMSE-percent
* MSE-percent should be square of RMSE-percent?

**Review by fkiraly — CHANGES_REQUESTED** [review]
@jusssch, I wanted to push some fixes, but I cannot push to your branch.

Could you kindly allow maintainers to push? That requires checking a box on the PR, of the same name.

**Comment by jusssch** [comment]
Sure! It should be working now, thanks!

**Pull request merged**